Source/WebCore/platform/graphics/ColorBlending.cpp

```diff
@@ -91,41 +91,63 @@ Color blendWithWhite(const Color& color)
     return result;
 }
 
+static bool requiresLegacyInterpolationRules(const Color& color)
+{
+    return color.callOnUnderlyingType([&] (const auto& underlyingColor) {
+        using ColorType = std::decay_t<decltype(underlyingColor)>;
+
+        if constexpr (std::is_same_v<ColorType, SRGBA<uint8_t>>)
+            return true;
+        else if constexpr (std::is_same_v<ColorType, SRGBA<float>>)
+            return true;
+        else if constexpr (std::is_same_v<ColorType, HSLA<float>>)
+            return true;
+        else if constexpr (std::is_same_v<ColorType, HWBA<float>>)
+            return true;
+        else if constexpr (std::is_same_v<ColorType, ExtendedSRGBA<float>>)
+            return !color.usesColorFunctionSerialization();
+        else
+            return false;
+    });
```

Comment:
Couldn't this sort of thing achieved without all the ifs and std::is_same_vs with WTF::switchOn() type approach?

Reply:
Yeah, an alternate approach would look like this:

```c++
    return color.callOnUnderlyingType(WTF::makeVisitor(
        [] (const SRGBA<uint8_t>&) -> bool {
            return true;
        },
        [] (const SRGBA<float>&) -> bool {
            return true;
        },
        [] (const HSLA<float>&) -> bool {
            return true;
        },
        [] (const HWBA<float>&) -> bool {
            return true;
        },
        [&] (const ExtendedSRGBA<float>&) -> bool {
            return !color.usesColorFunctionSerialization();
        },
        [] (const auto&) -> bool {
            return false;
        }
    ));
```

Do you have a preference for one vs. another?

Reply:
I think the latter looks nicer, requiring no decay_t<> etc.